pkg/nsx/services/securitypolicy/firewall.go

```diff
@@ -656,10 +656,18 @@ func (service *SecurityPolicyService) deleteSecurityPolicy(obj interface{}, isVp
 	finalSecurityPolicyCopy := *nsxSecurityPolicy
 	finalSecurityPolicyCopy.Rules = nsxSecurityPolicy.Rules
 
-	if isVpcEnabled(service) || isVpcCleanup {
-		vpcInfo, err := service.getVpcInfo(spNameSpace)
-		if err != nil {
-			return err
+	if isVpcEnabled(service) || isGcOrVpcCleanup {
+		var vpcInfo *common.VPCResourceInfo
+		if isGcOrVpcCleanup == false {
+			vpcInfo, err = service.getVpcInfo(spNameSpace)
+			if err != nil {
+				return err
+			}
+		} else {
+			// In GC or cleanup process, vpcInfo should be listed directly from security policy store to avoid calling VPC service.
+			// Get orgId, projectId, vpcId from security policy path "/orgs/<orgId>/projects/<projectId>/vpcs/<vpcId>/security-policies/<spId>"
+			vpcInfo = new(common.VPCResourceInfo)
+			vpcInfo.OrgID, vpcInfo.ProjectID, vpcInfo.VPCID, _ = nsxutil.ParseVPCPath(*(nsxSecurityPolicy.Path))
```

Comment:
Better check if Path is valid, or it may crash. Better to refactor ParseVPCPath to return err.

Reply:
Thanks for advice, It's good to check nsxSecurityPolicy.Path. but I feel that there is no need to check ParseVPCPath since we don't know what info the user needs. Also, if there is invalid info from VPC path, 
and later, API call or something will pop up error.